src/ui_ng/lib/src/list-replication-rule/list-replication-rule.component.ts

```diff
@@ -176,38 +173,42 @@ export class ListReplicationRuleComponent implements OnInit, OnChanges {
     }
   }
 
+  replicateRule(rules: ReplicationRule[]): void {
+    this.replicateManual.emit(rules);
+  }
+
   deletionConfirm(message: ConfirmationAcknowledgement) {
     if (message &&
       message.source === ConfirmationTargets.POLICY &&
       message.state === ConfirmationState.CONFIRMED) {
-      let rule: ReplicationRule = message.data;
-      toPromise<any>(this.replicationService
-        .deleteReplicationRule(rule.id))
-        .then(() => {
-          this.translateService.get('BATCH.DELETED_SUCCESS')
-            .subscribe(res => {
-              this.batchDelectionInfos[0] = BathInfoChanges(this.batchDelectionInfos[0], res);
-            });
-          this.reload.emit(true);
-        })
-        .catch(error => {
-          if (error && error.status === 412) {
-            Observable.forkJoin(this.translateService.get('BATCH.DELETED_FAILURE'),
-                this.translateService.get('REPLICATION.FAILED_TO_DELETE_POLICY_ENABLED')).subscribe(res => {
-              this.batchDelectionInfos[0] = BathInfoChanges(this.batchDelectionInfos[0], res[0], false, true, res[1]);
-            });
-          } else {
-            this.translateService.get('BATCH.DELETED_FAILURE').subscribe(res => {
-              this.batchDelectionInfos[0] = BathInfoChanges(this.batchDelectionInfos[0], res, false, true);
-            });
-          }
-        });
+      this.deleteOpe(message.data);
+    }
+  }
+
+  selectedChange(): void {
+    if (this.selectedRow.length === 1 && this.selectedRow[0].id === this.selectedId) {
+      this.hasJobs.emit(true);
     }
+
+    if (this.selectedRow.length === 1 && this.selectedRow[0].id !== this.selectedId) {
+      this.hasJobs.emit(false);
+    }
+
+    if (this.selectedRow.length > 1) {
+      this.hasJobs.emit(false);
+    }
+    let hnd = setInterval(() => this.ref.markForCheck(), 200);
+    setTimeout(() => clearInterval(hnd), 2000);
   }
 
   selectRule(rule: ReplicationRule): void {
-    this.selectedId = rule.id || '';
-    this.selectOne.emit(rule);
+    if (this.selectedRow.length === 0 || (this.selectedRow.length === 1 && this.selectedRow[0].id === rule.id)) {
```

Comment:
click columns: uncheck all the checkboxes; visible job viewer; show job list
checkbox checked: hide job viewer